.github/workflows/typos.yml

```diff
@@ -0,0 +1,15 @@
+name: Spell Check 
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
```

Comment:
Wouldn't we want it to run on PRs and not the `main` branch?

Reply:
I can confirm that the workflow is indeed configured to run on pull requests. The `pull_request` trigger in our workflow file ensures that the spell check will run for every new or updated pull request, in addition to the `push` events on the main branch. 

Please let me know if there are any other aspects you would like me to address or if everything looks good to go.

Reply:
Thanks! This is a great improvement.